applications/serial_lte_modem/overlay-native_tls.conf

```diff
@@ -1,61 +1,80 @@
-# To build serial LTE modem native TLS socket support
-# add the following to your west build command:
-#       -- -DEXTRA_CONF_FILE=overlay-native_tls.conf
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-# TLS configuration
 CONFIG_SLM_NATIVE_TLS=y
-CONFIG_MODEM_KEY_MGMT=y
+# Enable legacy crypto support
+CONFIG_NRF_SECURITY=y
 CONFIG_NORDIC_SECURITY_BACKEND=y
-CONFIG_NET_SOCKETS_SOCKOPT_TLS=y
-CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS=2
-CONFIG_NET_SOCKETS_TLS_SET_MAX_FRAGMENT_LENGTH=y
-# Increase extra FD entry for TLS contexts(2)
-CONFIG_POSIX_MAX_FDS=10
-# Enable Socket Logging for debug
-#CONFIG_NET_LOG=y
-#CONFIG_NET_SOCKETS_LOG_LEVEL_DBG=y
+CONFIG_MBEDTLS_PSA_CRYPTO_C=y
+# Enable TLS client and server via MBEDTLS library
 CONFIG_MBEDTLS=y
 CONFIG_MBEDTLS_TLS_LIBRARY=y
-CONFIG_MBEDTLS_ENABLE_HEAP=y
-# If larger TLS buffer is required for large CA chain,
-# increase CONFIG_MBEDTLS_SSL_IN_CONTENT_LEN and
-# CONFIG_MBEDTLS_SSL_OUT_CONTENT_LEN to 4096
-# and CONFIG_MBEDTLS_HEAP_SIZE to 36864
-CONFIG_MBEDTLS_SSL_IN_CONTENT_LEN=5120
+CONFIG_MBEDTLS_SSL_SRV_C=y
+CONFIG_MBEDTLS_CIPHER=y
+# Enable loading the modem with TLS certificates and keys.
+CONFIG_MODEM_KEY_MGMT=y
+# Set maximum TLS contexts.
+# Increase also CONFIG_POSIX_MAX_FDS, if you need more TLS contexts.
+CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS=2
+CONFIG_POSIX_MAX_FDS=10
```

Comment:
Not exactly brought by your PR, but I just noticed that there is the following in `slm_defines.h`:
```#define SLM_MAX_SOCKET_COUNT 8    /** re-define NRF_MODEM_MAX_SOCKET_COUNT */```
I think that this macro should be removed (or updated) and replaced by `CONFIG_POSIX_MAX_FDS` (defined in `nrf_socket.h`).

Actually, this seems especially relevant as here you are defining a maximum FD count that is different from `NRF_MODEM_MAX_SOCKET_COUNT`.